### ISSUE TITLE
Multiple bin in saved model

### ISSUE THREAD

**Issue by tryanbot** [facebookresearch/wav2letter#890]
Hi, I found multiple bin in the model folder. there is :
1. 001_model_last.bin
2. 001_model_[name_of_eval_lst_file].bin
could you please tell me what each bin means? coz based on my understanding, there should be only 1 model. why we have bin for each validation list?

**Comment by abhinavkulkarni** [comment]
@tryanbot: There are multiple validation sets (`dev-clean` and `dev-other`). The model selection is done based on those two independently. Based on your use case, you may choose to use one model or the other.

**Comment by tryanbot** [comment]
how the model selection works exactly? because it seems the 001_model_[name_of_eval_lst_file].bin is very much overfit with [name_of_eval_lst_file] , if I tried using audio file outside the list, the performance is drop quite significant. please explain more @abhinavkulkarni

**Comment by tlikhomanenko** [comment]
model_last - is the last model state, while all others are the best performing snapshots on each particular validation set. The metric is Viterbi WER, so we are tracking for each snapshot for each validation set the current Viterbi WER storing the best ones.

**Comment by tryanbot** [comment]
@tlikhomanenko : so there is exist one point of the training process such that model_last is equal to the 001_model_[name_of_eval_lst_file].bin ? because it seem the model_last give much worse performance, so much worse

**Comment by tlikhomanenko** [comment]
Yep. you have reportiters or otherwise every epoch we call evaluation on validation and storing the best checkpoint on validation, while model_last just rewriting the file every reportiters / epoch.

**Comment by tryanbot** [comment]
@tlikhomanenko : thanks, u re the most helpful. hv a great day

**Comment by tlikhomanenko** [comment]
You are welcome!
